airflow/contrib/operators/sqoop_operator.py

```diff
@@ -17,75 +17,160 @@
         """
         super(SqoopOperator, self).__init__(*args, **kwargs)
         self.conn_id = conn_id
-        self.type_cmd = type_cmd
+        self.cmd_type = cmd_type
         self.table = table
+        self.query = query
         self.target_dir = target_dir
         self.append = append
-        self.type = type
+        self.file_type = file_type
         self.columns = columns
         self.num_mappers = num_mappers
         self.split_by = split_by
         self.where = where
         self.export_dir = export_dir
+        self.input_null_string = input_null_string
+        self.input_null_non_string = input_null_non_string
+        self.staging_table = staging_table
+        self.clear_staging_table = clear_staging_table
+        self.enclosed_by = enclosed_by
+        self.escaped_by = escaped_by
+        self.input_fields_terminated_by = input_fields_terminated_by
+        self.input_lines_terminated_by = input_lines_terminated_by
+        self.input_optionally_enclosed_by = input_optionally_enclosed_by
+        self.batch = batch
+        self.direct = direct
+        self.driver = driver
+        self.verbose = verbose
+        self.relaxed_isolation = relaxed_isolation
+        # No mutable types in the default parameters
+        if properties is None:
+            properties = {}
+        self.properties = properties
 
     def execute(self, context):
         """
         Execute sqoop job
         """
-        hook = SqoopHook(conn_id=self.conn_id)
+        hook = SqoopHook(conn_id=self.conn_id,
+                         verbose=self.verbose,
+                         num_mappers=self.num_mappers,
+                         properties=self.properties)
 
-        if self.type_cmd is 'export':
+        if self.cmd_type is 'export':
             hook.export_table(
                 table=self.table,
                 export_dir=self.export_dir,
-                num_mappers=self.num_mappers)
+                input_null_string=self.input_null_string,
+                input_null_non_string=self.input_null_non_string,
+                staging_table=self.staging_table,
+                clear_staging_table=self.clear_staging_table,
+                enclosed_by=self.enclosed_by,
+                escaped_by=self.escaped_by,
+                input_fields_terminated_by=self.input_fields_terminated_by,
+                input_lines_terminated_by=self.input_lines_terminated_by,
+                input_optionally_enclosed_by=self.input_optionally_enclosed_by,
+                batch=self.batch,
+                relaxed_isolation=self.relaxed_isolation)
         else:
-            hook.import_table(
-                table=self.table,
-                target_dir=self.target_dir,
-                append=self.append,
-                type=self.type,
-                columns=self.columns,
-                num_mappers=self.num_mappers,
-                split_by=self.split_by,
-                where=self.where)
+            if not self.table:
```

Comment:
this is where I wanted to ask you most about. Why do you have the `not` in this statement? I could be misunderstanding but if someone passes the table don't we want to do the `import_table` function, and same in the query below statement

Reply:
I can see how this is misleading, feel free to rewrite.